### ISSUE TITLE
MayaUSDExportCommand - overzealous conflict checking when stripping namespaces and specifying roots

### ISSUE THREAD

**Issue by bentoogood** [Autodesk/maya-usd#1817]
**Describe the bug**
When specifying 'exportRoots' and 'stripNamespaces' as arguments for the mayaUSDExport command, prim path conflict checking is overzealous and prevents individual export of multiple references of the same asset.
Conflict checks should only be applied to the scene branches specified by the exportRoots argument, not across the entire Maya scene.

**Steps to reproduce**
1. Start a new blank Maya scene.
2. Run this command to create a test scene containing two spheres with unique names defined by their namespaces:
```
pm.namespace(add="myAssetA")
pm.polySphere(name="myAssetA:sphere_GEO")
pm.namespace(add="myAssetB")
pm.polySphere(name="myAssetB:sphere_GEO")
```
3. Run this command to export one of the spheres, but keeping the namespaces:
```
pm.mayaUSDExport(file="E:/root.usda",
exportRoots=['|myAssetA:sphere_GEO'],
)
```
4. Then run this command to attempt to export the same one sphere but with the namespace stripped:
```
pm.mayaUSDExport(file="E:/root_and_stripped.usda",
exportRoots=['|myAssetA:sphere_GEO'],
stripNamespaces=True,
)
```
This raises the error:

> # Error: RuntimeError: Multiple dag nodes map to the same prim path after stripping namespaces: |myAssetA:sphere_GEO - |myAssetB:sphere_GEO # 

**Expected behavior**
In this example case, there is no actual clash of dag nodes to prim paths, as there should only be the one Maya item targeted to the one USD prim.
The exporter should only check for conflicts within the branch(s) specified by exportRoots.

In this example case I would expect the exported prim to appear in the USD as:
```
#usda 1.0
(
    defaultPrim = "sphere_GEO"
    metersPerUnit = 0.01
    upAxis = "Y"
)

def Mesh "sphere_GEO" (
    prepend apiSchemas = ["MaterialBindingAPI"]
    kind = "component"
)
{
...
}

```

**Specs (if applicable):**
 - Windows 10
 - Maya 20202_U4
 - USD plugin 0.13.0_202110191046-3a94a0f

**Comment by santosg87** [comment]
I am also able to reproduce this on 2022.2 and 0.13.0 builds.

**Comment by icyvapor** [comment]
Just saw this in Maya 2023 with MayaUSD v0.18.0. 
Upvote.

**Comment by daviddeacon** [comment]
Yeah I bumped into this a little while ago, a little disappointing that it’s been on the back log for so long. 

My inelegant brute force work around was to export with out striping the namespaces and then using the usd python api to open the exported file and remove the namespaces manually…

**Comment by icyvapor** [comment]
Ouch, seems the USD export will convert ':' to '_' 
This is troublesome for us since there are usually some '_' in the naming convention.. 

At the moment i'm leaning towards use the `file` command with "USD Export" type.. which seems to be the underlying mechanism for the GUI export. Although I find it tricky to translate it from MEL To the Python version, if possible at all.

**Comment by bentoogood** [comment]
Has anyone been able to test with the new v0.19.0 release? I've not see any commits or PRs that might have addressed, but we're getting on for a year since reporting and it continues to be a problem for us so fingers crossed it's on the agenda if it's not resolved in 0.19

**Comment by tanant** [comment]
@bentoogood it's still outstanding on our schedule - didn't make it into the .19 release as we were focussing on things around the maya ref workflow (amongst other things) but we've recently called it out here internally to see if we can squeeze it into a release coming up.

**Comment by daviddeacon** [comment]
@tanant Thanks for calling this one out. I'm hoping you're able to get back on a monthly release schedule. The gap between .18 and .19 was a bit too long, and the issues in backlog list is getting full. 

I'm excited to see the  maya ref workflow, just hoping for some of the bug/issues to be squashed in the mean time.

Cheers,
David.

**Comment by bentoogood** [comment]
thanks @tanant - just hit this again this morning so fingers crossed for 0.20.0 !

**Comment by bentoogood** [comment]
Looks like this was addressed in 0.20.0 - at least the simple test case appears to run fine now. 
thanks for picking up.

**Comment by slyped** [comment]
Hmm, I just encountered this bug still today on 0.20.0 ..
